### PR TITLE
feat: add Span.set_tag_str

### PR DESCRIPTION
**Pull request by avara1986**
## Description
Rename `_set_str_tag` to `set_tag_str`. Span instances use `_set_str_tag` at many points, it's more consistent make this method public

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: ddtrace/span.py [hunk 1]
```diff
@@ -332,7 +332,7 @@ def set_tag(self, key, value=None):
         except Exception:
             log.warning("error setting tag %s, ignoring it", key, exc_info=True)
 
-    def _set_str_tag(self, key, value):
+    def set_str_tag(self, key, value):
```
Comment (Kyle-Verhoog) [inline ddtrace/span.py:335]:
```suggestion
    def set_tag_str(self, key, value):
```

maybe is a little better since we're changing it 😄

**Review by Kyle-Verhoog — COMMENTED** [review]
I'm good with making the method public, what's the motivation behind the change? Can we add it to the PR description?

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by gnufede — DISMISSED** [review]

**Review by juanjux — DISMISSED** [review]

**Review by majorgreys — CHANGES_REQUESTED** [review]
Can we keep the internal api as it is and use `_set_str_tag`? We can then backport for patch releases. The feature PR to make the internal interface public would be done as a follow up.

**Review by gnufede — DISMISSED** [review]

**Comment by majorgreys** [comment]
Thanks @avara1986!

This is a non-breaking feature change to the public interface. Can you update the PR title to reflect that and add a release note?

**Review by juanjux — DISMISSED** [review]

**Review by gnufede — APPROVED** [review]

File: releasenotes/notes/span-_set_str_tag-is-replaced-by-span-set_tag_str-271b7829efd9e6ac.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
```
Comment (majorgreys) [inline releasenotes/notes/span-_set_str_tag-is-replaced-by-span-set_tag_str-271b7829efd9e6ac.yaml:2]:
```suggestion
feature:
```

File: releasenotes/notes/span-_set_str_tag-is-replaced-by-span-set_tag_str-271b7829efd9e6ac.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  - |
+    ``Span._set_str_tag()`` is replaced by ``Span.set_tag_str()``.
```
Comment (majorgreys) [inline releasenotes/notes/span-_set_str_tag-is-replaced-by-span-set_tag_str-271b7829efd9e6ac.yaml:4]:
```suggestion
    Add ``Span.set_tag_str()`` as an alternative to the overloaded functionality of ``Span.set_tag()` when the value can be coerced to unicode text. This function can be used for performance optimizations in manual instrumentation.
```

**Review by majorgreys — CHANGES_REQUESTED** [review]
Release notes updated to reflect change to public API.

**Review by majorgreys — DISMISSED** [review]

**Review by gnufede — APPROVED** [review]

**Review by juanjux — APPROVED** [review]

**Pull request merged**